tests/foreman/api/test_reporttemplates.py

```diff
@@ -38,6 +39,52 @@
 from robottelo.vm import VirtualMachine
 
 
+@pytest.fixture(scope='class')
+def setup_content(request):
+    """"""
+    org = entities.Organization().create()
+    with manifests.clone() as manifest:
+        upload_manifest(org.id, manifest.content)
+    rh_repo_id = enable_rhrepo_and_fetchid(
+        basearch='x86_64',
+        org_id=org.id,
+        product=PRDS['rhel'],
+        repo=REPOS['rhst7']['name'],
+        reposet=REPOSET['rhst7'],
+        releasever=None,
+    )
+    rh_repo = entities.Repository(id=rh_repo_id).read()
+    rh_repo.sync()
+    custom_repo = entities.Repository(
+        product=entities.Product(organization=org).create(),
+    ).create()
+    custom_repo.sync()
+    lce = entities.LifecycleEnvironment(organization=org).create()
+    cv = entities.ContentView(
+        organization=org,
+        repository=[rh_repo_id, custom_repo.id],
+    ).create()
+    cv.publish()
+    cvv = cv.read().version[0].read()
+    promote(cvv, lce.id)
+    ak = entities.ActivationKey(
+        content_view=cv,
+        max_hosts=100,
+        organization=org,
+        environment=lce,
+        auto_attach=True
+    ).create()
+    subscription = entities.Subscription(organization=org).search(
+        query={'search': 'name="{}"'.format(DEFAULT_SUBSCRIPTION_NAME)})[0]
+    ak.add_subscriptions(data={
+        'quantity': 1,
+        'subscription_id': subscription.id,
+    })
+    request.cls.org_setup = org
+    request.cls.ak_setup = ak
+
+
+@pytest.mark.usefixtures("setup_content")
```

Comment:
Right now if I run `test_positive_CRUDL`, it will first execute your fixture - even though this test doesn't use anything from fixture.

This matters because your test is tier3, so it will be invoked in separate `pytest` run by automation nad won't leverage fixture that run for tier1. Also, during failure analysis we might run any single test in isolation.

I think it's better to move that decorator to `test_positive_generate_entitlements_report` method. This way this test will be the only one that pulls in fixture - which is good, because it's only one that makes actual use of it.

If tests are gradually ported to make use of fixture, they will need to have decorator, too. At some point we will just remove individual methods decorators and slap that on top of class.